src/session_pool/session.js

```diff
@@ -1,14 +1,48 @@
 import { cryptoRandomObjectId } from 'apify-shared/utilities';
 import log from 'apify-shared/log';
 import { checkParamOrThrow } from 'apify-client/build/utils';
-import tough from 'tough-cookie';
-import EVENTS from './events';
+import { Cookie, CookieJar } from 'tough-cookie';
+import { Cookie as PuppeteerCookie, SameSiteSetting } from 'puppeteer'; // eslint-disable-line import/named,no-unused-vars
+import { EventEmitter } from 'events'; // eslint-disable-line no-unused-vars
+import EVENTS from './events'; // eslint-disable-line import/named,no-unused-vars
 import { STATUS_CODES_BLOCKED } from '../constants';
 import { getCookiesFromResponse } from './session_utils';
 
-const { Cookie } = tough;
-
+/**
+ * Peristable {Session} state.
+ * @typedef {Object} SessionState
+ * @property {String} id
+ * @property {Object} cookies
+ * @property {CookieJar} cookieJar
+ * @property {Object} userData
+ * @property {Number} maxErrorScore
+ * @property {Number} errorScoreDecrement
+ * @property {Date} expiresAt
+ * @property {Date} createdAt
+ * @property {Number} usageCount
+ * @property {Number} errorScore
+ */
 
+/**
+ * @typedef {Object} SessionOptions
+ * @property [id] {String} - Id of session used for generating fingerprints. It is used as proxy session name.
+ * @property [maxAgeSecs=3000] {Number} - Number of seconds after which the session is considered as expired.
+ * @property userData {Object} - Object where custom user data can be stored. For example custom headers.
+ * @property [maxErrorScore=3] {number} - Maximum number of marking session as blocked usage.
+ *   If the `errorScore` reaches the `maxErrorScore` session is marked as block and it is thrown away.
+ *   It starts at 0. Calling the `markBad` function increases the `errorScore` by 1.
+ *   Calling the `markGood` will decrease the `errorScore` by `errorScoreDecrement`
+ * @property [errorScoreDecrement=0.5] {number} - It is used for healing the session.
+ *   For example: if your session is marked bad two times, but it is successful on the third attempt it's errorScore is decremented by this
+ *   number.
+ * @property createdAt {Date} - Date of creation.
```

Comment:
[createdAt]

Reply:
Done. It seems `expiresAt` also has a default value.